python/cudf/cudf/core/column/column.py

```diff
@@ -395,16 +395,23 @@ def view(self, dtype):
             The dtype to view the data as
         
         """
-        if self.null_count > 0:
-            raise ValueError("Can not produce a view of a column with nulls")
         dtype = np.dtype(dtype)
+
+        if self.null_count > 0:
+            if self.dtype.itemsize != dtype.itemsize:
+                raise ValueError("Can not produce a view of a column with nulls")
+            else:
+                pass
+
+
         if dtype.kind in ('o', 'u', 's'):
             raise TypeError('Bytes viewed as str without metadata is ambiguous')
+
         if self.data.size % dtype.itemsize:
             raise TypeError(
                 "Source data size must be an even multiple of view data size"
             )
-        return as_column(self.data, dtype=dtype)
+        return build_column(self.data, dtype=dtype, mask=self.mask)
```

Comment:
Should use `self.base_data`, `self.base_mask`, `self.offset`, and `self.size` here instead to avoid potential unnecessary copy construction.

Reply:
In the process of doing this I realized a case that throws a bit of a wrench in things. If I'm understanding correctly, what we want to do avoid copying by interpreting the view as the base column reinterpreted as the new datatype, as well as by scaling the offset and size according to the ratio of the old to new dtypes. But what do we do when the slice we're trying to view divides evenly into the new datatype, but the underlying base column does not? For example what if we had `col=[1,2,3,4,5]` with `int32` dtype, and we wanted to do something like `col[1:].view('int64')`? Now the fact that we're trying to construct the view as the same base column plus a different offset and size won't work even though logically what we're asking for should work.

Reply:
When you say logically, you mean without respect to the underlying memory representation?

Reply:
So given this is only safe if there's no nulls anyway, we can just calculate a new offset based on the change in dtype size.

Reply:
But that might not be a valid number, right? I mean that viewing the last four `int32` type elements of a five element array as `int64` should be able to give me back a valid length two array of `int64`s. But if we try and do this by first converting the base five element array into `int64` and then just scaling the offset it won't work, the offset would be 0.5.

Reply:
I think the only way to do this without a copy in any scenario is to do the actual pointer math to find out where the new data needs to actually start and construct a buffer around that integer.

Reply:
I think we need to split this into two different paths:
1) dtype size stays the same: We can take the `base_*` approach to avoid copy constructing a new mask unnecessarily.
2) dtype size changes: since we already guaranteed there's no nulls, we can just use pointer arithmetic to point in the data buffer properly

Reply:
That sounds like the way forward 👍 .

Reply:
As null columns are not supported in case of different dtype sizes, we can do this. But for same dtype sizes, we need to use the same offset.

Reply:
Are there any gotchas with this approach with respect to overlapping buffers, like w.r.t memory ownership, etc? Suppose I have an array `ary` and a view `view=ary[n:m].view(to_dtype)`, under which there is a different buffer (by python `id`). What happens when `ary` goes out of scope, is `view` still valid?

Reply:
If you set the `owner` to the `Buffer` we handle reaching inside that `Buffer` and setting the owner appropriately in the Buffer `__init__`